docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
 
-:::warning
-If you want to use a different GitHub account for BobaBoard, your `origin` URL
-will be different. Follow the instructions in the ["use a different GitHub account" section](#extra-use-a-different-github-account).
+:::info
+The name of your fork doesn't have to match the name of the original project. For this guide, however, we'll assume you have kept the same name.
 :::
 
-At this point, you can proceed with the regular flow for a pull request.
+### 2. Clone your fork to your machine
 
-### 1. Create a branch off of `master` for your new feature
+This means copying the code that is stored in your GitHub account to your PC. It doesn't really matter where you want to do this (make sure you have permissions?), though you might want to create a folder to keep all your BobaBoard-related coding in. For example: `/home/NAME/code-projects/bobaboard`.
 
-In the folder for your **forked** repository (see step 0) run
+:::danger
+
+Windows users using WSL will want to pay extra attention.
+
+> //TODO: be less mysterious about it...
+
+:::
+
+In order to clone the repository you need to tell git where the code you want to copy is stored. Simply put, you need a URL. Your repository's URL is simply the web address you use to view it on GitHub with a `.git` at the end. Meaning that if the URL you can copy from your browser's navigation bar is: `https://github.com/<YOUR_GITHUB_USERNAME>/boba-components` then the URL you'll need to use when cloning is:  `https://github.com/<YOUR_GITHUB_USERNAME>/boba-components.git`.
+
+GitHub actually has a handy button in its interface to get this URL.
+
+> //TODO: add screenshot?
+
+:::caution On HTTPS vs SSH
+There's HTTPS and SSH..?
+:::
+
+When you run the clone command a new folder named after the repository will be created and all the files will be stored inside. Navigate to the folder where you want to store the project and then run the clone command.
 
 ```bash   showLineNumbers
-git branch -f new-feature-name
-git checkout new-feature-name
+cd /home/<USER>/code-projects/bobaboard
+git clone <REPOSITORY_URL>
+```
+
+After cloning a few boba-repos and making a few personal projects you might end up with a folder structure that looks something like this:
+
+```none
+/home
+├─ /<USER_NAME>
+│  ├─ code-projects/
+│  │  ├─ bobaboard/
+│  │  │  ├─ boba-components/
+│  │  │  ├─ boba-docs/
+│  │  │  ├─ boba-editor/
+│  │  ├─ boba-tan-shrine/
+│  │  ├─ personal-website/
 ```
 
-Of course you can swap `new-feature-name` with whatever string represents what you're trying to do.
+One thing to keep in mind is that the code of a project can change fairly often, meaning you'll want to keep your fork in sync with the original repository, and your clone in sync with your fork. To do this you need to make sure that the local copy of the code that lives in your machine correctly points to both of these GitHub repositories (the one that belong to the BobaBoard organization, and the one in your own account).
```

Comment:
I would figure out how to move the sync'ing later. At least we can do a callout like "You'll need to keep the code in sync, read section [x] later to see how to do it".

Reply:
i fucking did it aaa